src/lib/definition-parser.ts

```diff
@@ -282,8 +282,8 @@ async function calculateDependencies(
 		}
 		const pathMapping = pathMappingList[0];
 
-		// Path mapping may be for "@foo/bar" -> "foo__bar".
-		const scopedPackageName = unmangleScopedPackage(pathMapping);
+		// Path mapping may be for "@foo/bar" -> "foo__bar" or "@foo/bar" -> "foo__bar/v2"
+		const scopedPackageName = unmangleScopedPackage(stripVersion(pathMapping));
 		if (scopedPackageName !== undefined) {
 			if (dependencyName !== scopedPackageName) {
 				throw new Error(`Expected directory ${pathMapping} to be the path mapping for ${dependencyName}`);

```

Comment:
`calculateDependencies` is odd in that only versioned packages need an entry in the dependencies it calculates. Previously, since we didn't check scoped packages for versions, the `continue` below was fine.

However, if you want to correctly handle versioned, scoped packages, then you'll have to move this section below the version-checking code that starts with `const majorVersion = ...` on line 294. Then that section will need to handle scoped packages correctly. I guess that the handling will be something like `{ name: scopedPackageName, majorVersion }`, where `scopedPackageName = unmangleScopedPackage(stripVersion(pathMapping))` (like you have above) and `majorVersion = parseDependencyVersionFromPath(...`. But you'll have to make sure that parseDependencyVersionFromPath handles scoped packages as well, perhaps by calling it with different arguments for the scoped version case.